buildenv/jenkins/common/variables-functions

```diff
@@ -327,4 +334,25 @@ def set_job_variables(job_type) {
     }
 }
 
+def validate_arguments(ARGS) {
+    for (arg in ARGS) {
+        if (!params.get(arg)) {
+            error("Must specify ${arg}")
+        }
+
+        if (arg == 'PLATFORM') {
+            SPEC = params.get(arg)
```

Comment:
Should probably add a comment about why we use SPEC and PLATFORM and the difference between a groovy variable and a Jenkins Param and why SPEC conflicts with a compile.

Reply:
comment added